File: Sem2/PO_project/c++/animals/antelope.cpp
#include "antelope.hpp"

Antelope::Antelope(int x, int y, Map* map)
    : Animal(x, y, 4, 4, map, "Antelope") {}

Antelope* Antelope::reproduce(int newX, int newY) {
    return new Antelope(newX, newY, map);
}

int Antelope::movementRange() const {
    return 2;
}

bool Antelope::tryToEscape(Organism* attacker) {
    if (rand() % 2 != 0) return false;

    int ex = x;
    int ey = y;
    if (map->findFreeSpaceAround(ex, ey)) {
        x = ex;
        y = ey;
        return true;
    }
    return false;
}

File: Sem2/PO_project/python/animals/antelope.py
from __future__ import annotations

from python.organism import Animal


class Antelope(Animal):
    def __init__(self, strength: int, initiative: int, x: int, y: int, game) -> None:
        super().__init__(strength, initiative, x, y, game)
        self.range = 2

    def get_color(self) -> tuple[int, int, int]:
        return (160, 82, 45)

    def get_organism(self):
        return self

    def reproduce(self, x: int, y: int):
        return Antelope(4, 4, x, y, self.game)
